### DISCUSSION TITLE
Preparing for Ember Data 5 and beyond

### DISCUSSION THREAD

**Discussion by roomman** [mikkopaderes/ember-cloud-firestore-adapter#270] [General]
I've been trying to follow the planned updates to Ember Data. I'm still not entirely clear on the implications for this and other add-ons that my projects depend on. Does anyone else have clarity on how the changes might impact this add-on in particular?

**Comment by mikkopaderes** [comment]
It's very confusing at this point for me. There's a lot of terms being thrown as well as APIs but I've yet to see an actual resource on how things will be looking moving forward for a regular ember-data consumer.

**Reply by roomman** [reply]
I'm glad it's not just me then 🤣

When you have clarity, if there's anything you need help with then I'm happy to contribute. 🙌🏻

**Comment by roomman** [comment]
@charlesfries I hope you are well. I see you've opened some PRs to prepare this addon for the next generation Ember Data. I am seeing the following deprecation warning:

```
DEPRECATION: Using Snapshot.type to access the ModelClass for a record is deprecated. Use store.modelFor(<modelName>) instead. [deprecation id: ember-data:deprecate-snapshot-model-class-access] This will be removed in ember-data 5.0.
```

I believe it relates to this [block of code](https://github.com/search?q=repo%3Amikkopaderes%2Fember-cloud-firestore-adapter%20Snapshot.type&type=code). Happy to open a PR for to but don't want to duplicate any WIP you might have. Are you happy for me to muck in here?

**Reply by charlesfries** [reply]
Go for it. This old PR has the change as well, if you want to use it as a reference (https://github.com/mikkopaderes/ember-cloud-firestore-adapter/pull/278/files)

**Reply by charlesfries** [reply]
Just submitted #286

**Reply by roomman** [reply]
Ace, thanks. Sorry I didn't get to it in time. Let me know if there's anything else I can do to help with updating this repo. Very happy to invest some time - I'm extra good at donkey work! 👍🏼

**Comment by mikkopaderes** [comment]
Hey all. Haven't been in touch with Ember for more than a year now so I'm just out of loop already with what's up with it. @charlesfries if you're keen for it, I can give you full rights to this repo or perhaps transfer it to [adopted ember addons](https://github.com/adopted-ember-addons) or maybe even to you yourself. What do you think?

**Reply by charlesfries** [reply]
Hey @mikkopaderes, any of those options will work! I think just having write access (and maybe the ability to publish releases) would be fine. Thank you!
